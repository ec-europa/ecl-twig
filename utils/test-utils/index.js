const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

module.exports = {
  renderTwigFile,
};
