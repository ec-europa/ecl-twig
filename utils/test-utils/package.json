{
  "name": "@ecl-twig/test-utils",
  "version": "0.0.0-test-only",
  "description": "A local test helper",
  "main": "index.js",
  "private": true
}
