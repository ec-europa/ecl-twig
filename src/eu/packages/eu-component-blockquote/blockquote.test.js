import path from 'path';
import { renderTwigFile } from '@ecl-twig/test-utils';

describe('EU - Blockquote', () => {
  test('renders correctly', done => {
    expect.assertions(1);

    const template = path.resolve(__dirname, './blockquote.html.twig');
    const data = {
      body: 'Example body',
      author: 'Author name',
    };

    renderTwigFile(template, data, (err, html) => {
      expect(html).toMatchSnapshot();
      done();
    });
  });
});
