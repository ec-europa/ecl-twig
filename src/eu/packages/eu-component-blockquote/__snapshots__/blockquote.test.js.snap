// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EU - Blockquote renders correctly 1`] = `
"<blockquote class=\\"ecl-blockquote\\">
  <p class=\\"ecl-blockquote__body\\"></p>
  <footer class=\\"ecl-blockquote__attribution\\">
    <cite class=\\"ecl-blockquote__author\\"></cite>
  </footer>
</blockquote>
"
`;
